### PR TITLE
[feat]: [PL-24061]: Authentication and Heartbeat support

### PR DESCRIPTION
**Pull request by mankrit-singh**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32331)

### CODE REVIEW

**Comment by mankrit-singh** [comment]
trigger messagecheck

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/adapter/AzureRepoToGitMapper.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import io.harness.delegate.beans.connector.scm.azurerepo.AzureRepoUsernameTokenDTO;
+import io.harness.delegate.beans.connector.scm.genericgitconnector.GitConfigDTO;
+import io.harness.encryption.SecretRefData;
+import io.harness.exception.InvalidRequestException;
+
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+@OwnedBy(HarnessTeam.PL)
+public class AzureRepoToGitMapper {
```
Comment (goelsatyam2) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/adapter/AzureRepoToGitMapper.java:26]: this function can be broken down into smaller chungs, using the separate functions for HTTP and SSH.
And you can use switch case here

File: 952-scm-java-client/src/main/java/io/harness/impl/scm/ScmGitProviderMapper.java [hunk 2]
```diff
@@ -56,12 +61,30 @@ public Provider mapToSCMGitProvider(ScmConnector scmConnector, boolean debug) {
...
+    } else if (scmConnector instanceof AzureRepoConnectorDTO) {
+      return mapToAzureRepoProvider((AzureRepoConnectorDTO) scmConnector, debug);
     } else {
       throw new NotImplementedException(
           String.format("The scm apis for the provider type %s is not supported", scmConnector.getClass()));
     }
   }
 
+  private Provider mapToAzureRepoProvider(AzureRepoConnectorDTO azureRepoConnector, boolean debug) {
+    String url = StringUtils.substringBeforeLast(azureRepoConnector.getUrl(), "/_git/");
```
Comment (goelsatyam2) [inline 952-scm-java-client/src/main/java/io/harness/impl/scm/ScmGitProviderMapper.java:73]: please make the constants for hardcoded values. And we can add the logic to how we are breaking the URL

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> Reviewed 2 of 12 files at r1, 6 of 9 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32331)* status: 14 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmGitProviderMapper.java, line 73 at r2](https://reviewable.io/reviews/harness/harness-core/32331#-N066_s429fXjeQWGEp6:-N066_s429fXjeQWGEp7:b-ix7goo) ([raw file](https://github.com/harness/harness-core/blob/0491fe08ac40ce6b9154948bae4357d9ac250e94/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmGitProviderMapper.java#L73)):*
> ```Java
> 
>   private Provider mapToAzureRepoProvider(AzureRepoConnectorDTO azureRepoConnector, boolean debug) {
>     String url = StringUtils.substringBeforeLast(azureRepoConnector.getUrl(), "/_git/");
> ```

Shouldn't it be .git ?

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32331)* status: 14 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmGitProviderMapper.java, line 76 at r2](https://reviewable.io/reviews/harness/harness-core/32331#-N066jRHDWKigPyyH4ry:-N066jRHDWKigPyyH4rz:bp8wpmd) ([raw file](https://github.com/harness/harness-core/blob/0491fe08ac40ce6b9154948bae4357d9ac250e94/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmGitProviderMapper.java#L76)):*
> ```Java
>     String orgAndProject = StringUtils.substringAfter(url, "dev.azure.com/");
>     String org = StringUtils.substringBefore(orgAndProject, "/");
>     String project = StringUtils.substringAfter(orgAndProject, "/");
> ```

Extract this logic into some function so that it can be used later also

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32331)* status: 14 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[952-scm-java-client/src/main/java/io/harness/impl/scm/ScmGitProviderMapper.java, line 81 at r2](https://reviewable.io/reviews/harness/harness-core/32331#-N066o9T3k6GbCZffLZ8:-N066o9T3k6GbCZffLZ9:bjoaqaf) ([raw file](https://github.com/harness/harness-core/blob/0491fe08ac40ce6b9154948bae4357d9ac250e94/952-scm-java-client/src/main/java/io/harness/impl/scm/ScmGitProviderMapper.java#L81)):*
> ```Java
>     AzureRepoApiAccessDTO apiAccess = azureRepoConnector.getApiAccess();
>     AzureRepoTokenSpecDTO bitbucketUsernameTokenApiAccessDTO = (AzureRepoTokenSpecDTO) apiAccess.getSpec();
>     String pat = String.valueOf(bitbucketUsernameTokenApiAccessDTO.getTokenRef().getDecryptedValue());
> ```

name it personalAccessToken to be more explicit

**Review by DeepakPatankar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32331)* status: 14 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/adapter/AzureRepoToGitMapper.java, line 44 at r2](https://reviewable.io/reviews/harness/harness-core/32331#-N01SKMlWPT9ORuBC8uM-r2-44:-N066yt5EaE2f9KZJJa4:biwp8ib) ([raw file](https://github.com/harness/harness-core/blob/0491fe08ac40ce6b9154948bae4357d9ac250e94/954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/adapter/AzureRepoToGitMapper.java#L44)):*
<details><summary><i>Previously, goelsatyam2 (Satyam Goel) wrote…</i></summary><blockquote>

why we are hardcoding the value of executing on delegate to true always?
</blockquote></details>

For Azure Repo, it will always be true as we don't support using this repo on manager

**Pull request closed without merging**